core/executor/src/wasm_executor.rs

```diff
@@ -679,6 +703,40 @@ impl WasmExecutor {
 		method: &str,
 		data: &[u8],
 	) -> Result<Vec<u8>> {
+		self.call_in_wasm_module_with_custom_signature(
+			ext,
+			module_instance,
+			method,
+			|alloc| {
+				let offset = alloc(data)?;
+				Ok(vec![I32(offset as i32), I32(data.len() as i32)])
```

Comment:
Is there any guarantee that `offset` is below `2**31` or is it just common sense that it's too expensive to allocate that much?

Reply:
WASM is only 32bit, so I think `offset` will always be below `2**31`.